### PR TITLE
feat: updated telemetry with tags

### PR DESCRIPTION
**Pull request by tobuck-aws**
### Description of how you validated changes

Added tests

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/awslabs/sbt-aws/blob/main/CONTRIBUTING.md)

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.*

### CODE REVIEW

File: src/control-plane/billing/billing-provider.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export class BillingProvider extends Construct {
    */
   constructor(scope: Construct, id: string, props: BillingProviderProps) {
     super(scope, id);
+    addTemplateTag(this, 'BillingProvider');
```
Comment (suhussai) [inline src/control-plane/billing/billing-provider.ts:57]: would have having the actual billing provider that the customer is using be more useful?

We could have implementers of IBilling specify a name (similar to `idpName`) and we just surface that here?

File: src/control-plane/control-plane-api.ts [hunk 2]
```diff
@@ -26,7 +27,7 @@ export class ControlPlaneAPI extends Construct {
   public readonly tenantUpdateServiceTarget: targets.ApiGateway;
   constructor(scope: Construct, id: string, props: ControlPlaneAPIProps) {
     super(scope, id);
-
+    addTemplateTag(this, 'ControlPlaneAPI');
```
Comment (suhussai) [inline src/control-plane/control-plane-api.ts:30]: The ControlPlaneAPI currently is tightly coupled with the ControlPlane such that you can't use one without the other, right? If that's the case, do we need this?

File: src/control-plane/services.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export class Services extends Construct {
 
   constructor(scope: Construct, id: string, props: ServicesProps) {
     super(scope, id);
+    addTemplateTag(this, 'Services');
```
Comment (suhussai) [inline src/control-plane/services.ts:24]: Same comment as for the ControlPlaneAPI tag.

File: src/utils/utils.ts [hunk 4]
```diff
@@ -4,8 +4,45 @@
 import { Stack } from 'aws-cdk-lib';
 import { Construct } from 'constructs';
 
-export const setTemplateDesc = (construct: Construct, sbtDesc: string) => {
+export const addTemplateTag = (construct: Construct, tag: string) => {
   const stackDesc = Stack.of(construct).templateOptions.description;
-  let description = stackDesc === undefined ? sbtDesc : stackDesc + ' - ' + sbtDesc;
+  const baseTelemetry = 'SaaS Builder Toolkit (uksb-1tupboc57)';
```
Comment (suhussai) [inline src/utils/utils.ts:9]: should we use the full name here? or would the smaller one be better to conserve space? (`sbt-aws`)

File: src/utils/utils.ts [hunk 5]
```diff
@@ -4,8 +4,45 @@
...
+    description = appendTagToDescription(stackDesc, tag);
+  }
   Stack.of(construct).templateOptions.description = description;
 };
+
+const appendTagToDescription = (existingDescription: string, newTag: string): string => {
+  // Check if the existing description already has tags
+  if (existingDescription.includes('(Tags:')) {
+    // Extract the existing tags
+    const startIndex = existingDescription.indexOf('(Tags:') + 6;
```
Comment (suhussai) [inline src/utils/utils.ts:30]: the `6` is throwing me off a little. Is it over the top to do something like this:

```
const tagOpeningString = '(Tags:';
const startIndex = existingDescription.indexOf(tagOpeningString) + tagOpeningString.length;
```

File: test/utils.test.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { App, Stack } from 'aws-cdk-lib';
+import { addTemplateTag } from '../src/utils';
+const app = new App();
+const telemetryConst = 'SaaS Builder Toolkit (uksb-1tupboc57)';
+
+describe('addTemplateTag', () => {
```
Comment (suhussai) [inline test/utils.test.ts:9]: Very nice!

**Review by suhussai — APPROVED** [review]
LGTM!

Minor things, but can be merged if urgent.

File: src/control-plane/services.ts (hunk 3 above)
Comment (tobuck-aws) [reply src/control-plane/services.ts:24]: Removed

File: src/control-plane/control-plane-api.ts (hunk 2 above)
Comment (tobuck-aws) [reply src/control-plane/control-plane-api.ts:30]: Removed

**Pull request merged**